Update event-driven architecture documentation to mention workflow patterns (#4210)
MicrosoftDocs/architecture-center: docs/guide/architecture-styles/event-driven-content.md

- Guaranteed delivery. In some systems, especially in IoT scenarios, it's crucial to guarantee that events are delivered.
- Processing events in order or exactly once. Each consumer type typically runs in multiple instances, for resiliency and scalability. This can create a challenge if the events must be processed in order (within a consumer type), or if the processing logic isn't idempotent.

> - Multiple events related to a single business process. [Workflow patterns](https://docs.particular.net/architecture/workflows/) can be used to reliably coordinate message flows across various services.

Reviewer: Can you re-phrase this?  In the prior two challenges, it starts with a brief sentence, and then the second sentence describes an example.  In yours, there is no description or example.

> - Multiple events related to a single business process. [Workflow patterns](https://docs.particular.net/architecture/workflows/) can be used to reliably coordinate message flows across various services.

Author: @ckittel I've tried to keep the style of the first sentence aligned with the other bullet points. Does the following suggestion align better:

```suggestion
- Coordinating messages across services. Business processes often involve multiple services publishing and subscribing messages to achieve a consistent outcome across a whole system. [Workflow patterns](https://docs.particular.net/architecture/workflows/) can be used to reliably manage message flows across various services.
```

> - Coordinating messages across services. Business processes often involve multiple services publishing and subscribing to messages to achieve a consistent outcome across a whole workload. [Workflow patterns](https://docs.particular.net/architecture/workflows) can be used to reliably manage message flows across various services.

Reviewer: ```suggestion
- Coordinating messages across services. Business processes often involve multiple services publishing and subscribing to messages to achieve a consistent outcome across a whole workload. [Workflow patterns](https://docs.particular.net/architecture/workflows) such as the [Choreography pattern](/azure/architecture/patterns/choreography) and [Saga Orchestration](/azure/architecture/reference-architectures/saga/saga#orchestration) can be used to reliably manage message flows across various services.
```

> - Coordinating messages across services. Business processes often involve multiple services publishing and subscribing to messages to achieve a consistent outcome across a whole workload. [Workflow patterns](https://docs.particular.net/architecture/workflows/) can be used to reliably manage message flows across various services.

Reviewer: This link doesn't work. [docs.particular.net/architecture/workflows](https://docs.particular.net/architecture/workflows/) results in a 404.  The `/` breaks it.

```suggestion
- Coordinating messages across services. Business processes often involve multiple services publishing and subscribing to messages to achieve a consistent outcome across a whole workload. [Workflow patterns](https://docs.particular.net/architecture/workflows) can be used to reliably manage message flows across various services.
```

Author: good catch, thanks ❤️
